Src/FluentAssertions/Data/Extensions/DataColumnCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System.Data;
+using System.Linq;
+
+using FluentAssertions.Collections;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions
+{
+    public static class DataColumnCollectionAssertionExtensions
+    {
+        /// <summary>
+        /// Assert that the current collection has the same number of elements as <paramref name="otherCollection" />.
+        /// </summary>
+        /// <param name="otherCollection">The other collection with the same expected number of elements</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataColumn>> HaveSameCount(this GenericCollectionAssertions<DataColumn> assertion, DataColumnCollection otherCollection, string because = "",
```

Comment:
Because `DataTableCollection`, `DataColumnCollection` and `DataRowCollection` don't implement the generic collection interfaces and thus don't match `HaveSameCount` in `GenericCollectionAssertions`, which can only accept an `IEnumerable<TExpectation>`.